src/analyzer/protocol/ssl/spicy/SSL.spicy

```diff
@@ -0,0 +1,2058 @@
+    server_key_exchange = 12,
+    certificate_request = 13,
+    server_hello_done = 14,
+    certificate_verify = 15,
+    client_key_exchange = 16,
+    finished = 20,
+    certificate_url = 21, # RFC 3546
+    certificate_status = 22, # RFC 3546
+};
+
+type SSL2ProtocolMessages = enum {
+    ssl_error = 0,
+    ssl_client_hello = 1,
+    ssl_client_master_key = 2,
+    ssl_client_finished = 3,
+    ssl_server_hello = 4,
+    ssl_server_verify = 5,
+    ssl_server_finished = 6,
+    ssl_request_certificate = 7,
+    ssl_client_certificate = 8,
+};
+
+type Extensions = enum {
+    server_name = 0,
+    max_fragment_length = 1,
+    client_certificate_url = 2,
+    trusted_ca_keys = 3,
+    truncated_hmac = 4,
+    status_request = 5,
+    user_mapping = 6,
+    client_authz = 7,
+    server_authz = 8,
+    cert_type = 9,
+    supported_groups = 10, # old name: elliptic_curves - draft-ietf-tls-negotiated-ff-dhe
+    ec_point_formats = 11,
+    srp = 12,
+    signature_algorithms = 13,
+    use_srtp = 14,
+    heartbeat = 15,
+    application_layer_protocol_negotiation = 16,
+    status_request_v2 = 17,
+    signed_certificate_timestamp = 18,
+    client_certificate_type = 19,
+    server_certificate_type = 20,
+    padding = 21,
+    encrypt_then_mac = 22,
+    extended_master_secret = 23,
+    token_binding = 24, # temporary till 2017-03-06 - draft-ietf-tokbind-negotiation
+    cached_info = 25,
+    SessionTicket_TLS = 35,
+    key_share_old = 40, # new for TLS 1.3, used in some of the drafts. Did not make it into the RFC. Was used for extended_random before.
+    pre_shared_key = 41, # new for 1.3, see RFC 8446
+    early_data = 42, # new for 1.3, see RFC 8446
+    supported_versions = 43, # new for 1.3, see RFC 8446
+    cookie = 44, # new for 1.3, see RFC 8446
+    psk_key_exchange_modes = 45, # new for 1.3, see RFC 8446
+    TicketEarlyDataInfo = 46, # new for 1.3 drafts, e.g. draft-16. Did not make it into the final RFC
+    certificate_authorities = 47, # new for 1.3, see RFC 8446
+    oid_filters = 48, # new for 1.3, see RFC 8446
+    post_handshake_auth = 49, # new for 1.3, see RFC 8446
+    signature_algorithms_cert = 50, # new for 1.3, see RFC 8446
+    key_share = 51, # new for 1.3, see RFC 8446
+    transparency_info = 52, # temporary - draft-ietf-trans-rfc6962-bis-34
+    connection_id = 53, # temporary -d draft-ietf-tls-dtls-connection-id
+    external_id_hash = 55, # temporary - RFC-ietf-mmusic-sdp-uks-07
+    external_session_id = 56, # temporary - RFC-ietf-mmusic-sdp-uks-07
+    quic_transport_parameters = 57, # temporary - draft-ietf-quic-tls-32
+    next_protocol_negotiation = 13172,
+    origin_bound_certificates = 13175,
+    encrypted_client_certificates = 13180,
+    channel_id = 30031,
+    channel_id_new = 30032,
+    padding_old = 35655,
+    renegotiation_info = 65281,
+};
+
+# type ECPointFormat = enum {
+#  uncompressed = 0,
+#  ansiX962_compressed_prime = 1,
+#  ansiX962_compressed_char2 =2
+# };
```

Comment:
There's a couple instances of commented out code here, not sure whether you left them in deliberately.

Reply:
Yup, I did. It's for referencing/potential future extensions. They are work to search for - and there does not seem to be any reason to not just keep them in there commented out.